packages/labs/ssr/src/test/lib/render-result_test.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {test} from 'uvu';
+// eslint-disable-next-line import/extensions
+import * as assert from 'uvu/assert';
+import {collectResult, collectResultSync} from '../../lib/render-result.js';
+
+test('collectResultSync collects strings', () => {
+  assert.equal(collectResultSync(['a', 'b', 'c']), 'abc');
+});
+
+test('collectResultSync collects strings and iterables', () => {
+  assert.equal(collectResultSync(['a', ['b', 'c'], 'd']), 'abcd');
+});
+
+test('collectResultSync throws for a Promise', () => {
+  assert.throws(
+    () => collectResultSync(['a', Promise.resolve('b'), 'c']),
+    'abc'
+  );
+});
+
+test('collectResult throws for iterables of Promises', async () => {
+  assert.throws(
+    () => collectResultSync(['a', [Promise.resolve('b')], 'c']),
+    'abc'
+  );
+});
+
+test('collectResult collects strings', async () => {
+  assert.equal(await collectResult(['a', 'b', 'c']), 'abc');
+});
+
+test('collectResultSync collects strings and iterables', async () => {
+  assert.equal(await collectResult(['a', ['b', 'c'], 'd']), 'abcd');
+});
+
+test('collectResult collects strings and Promises', async () => {
+  assert.equal(await collectResult(['a', Promise.resolve('b'), 'c']), 'abc');
+});
+
+test('collectResult collects strings and iterables of Promises', async () => {
+  assert.equal(await collectResult(['a', [Promise.resolve('b')], 'c']), 'abc');
+});
```

Comment:
I think we need a test of `['a', Promise.resolve(['b', 'c']), 'd']`, which I think is what a common case of a generator pausing and then resuming would look like?

And e.g. a double-nested one:`['a', Promise.resolve(['b', 'c', Promise.resolve(['d', 'e']), 'f']), 'g']`

Reply:
Actually, will a promised-value ever be a simple string rather than an iterable? I suppose it doesn't hurt, but I'm not following how we'd get that, since all of our "render" functions generally return iterables, not strings.

Reply:
I don't think a `Promise<string>` will not be common, but it's simple enough to support. I could remove it.

Reply:
Added tests and fixed the bugs!